### PR TITLE
[Quant] Support MXFP4 W4A16 for compressed-tensors MoE models 

### PR DESCRIPTION
**Pull request by dsikka**
## Purpose
Follow-up to:  https://github.com/vllm-project/vllm/pull/31926/#top
Add MoE support for MXFP4 CT Models. Currently only supporting weight-only quantization through the marlin integration.

There's a ton of shared boilerplate code between mxfp4 and nvfp4 that we should consider combining.

## Test Plan

Qwen3 MoE MXFP4A16: https://huggingface.co/nm-testing/Qwen3-30B-A3B-MXFP4A16/tree/main

## Test Result

```bash

Running GSM8K evaluation: 1319 questions, 5-shot
Evaluating: 100%|█████████████████████████████| 1319/1319 [00:29<00:00, 45.47it/s]

Results:
Accuracy: 0.882
Invalid responses: 0.000
Total latency: 29.023 s
Questions per second: 45.446
Total output tokens: 156720
Output tokens per second: 5399.780
```

### CODE REVIEW

**Review by mgoin — APPROVED** [review]
Looks clean to me, let's just get a test in and can land

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py [hunk 1]
```diff
@@ -224,6 +229,140 @@ def get_moe_method(
...
+    def create_weights(
+        self,
+        layer: torch.nn.Module,
+        num_experts: int,
+        hidden_size: int,
+        intermediate_size_per_partition: int,
+        params_dtype: torch.dtype,
+        **extra_weight_attrs,
+    ):
+        layer.num_experts = num_experts
```
[the remark this answers is not in the document]
Comment (dsikka) [reply vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py:248]: I dont think this is required?
Seems to be setting correctly

**Comment by dsikka** [comment]
> Looks clean to me, let's just get a test in and can land

Done. Can you add a ready label? @mgoin

**Review by mgoin — APPROVED** [review]
